### PR TITLE
Bugfix: Only allow 1 to N rank input for operators that take 1 axis

### LINKED ISSUE #873: Clarify 0D tensor (rank of 0) support for axis-dependent operations
The current WebNN spec doesn't clearly specify whether 0D tensors (scalars) are supported as inputs for operations that take an `axis` input/parameter.
Operations like `concat`, `split`, `cumulativeSum`, `batchNormalization`, etc. require an axis input/parameter, but 0D tensors have no dimensions to operate on.

Examples:
- `split(scalar, splits=2, axis=0)`
- `cumsum(scalar, axis=0)`
- `reduceSum(scalar, axes=[0])`

It would be helpful to specify whether some operations should handle 0D inputs with special rules or reject them (allowed ranks > 0).

### PR DESCRIPTION
**Pull request by huningxin**
Only allow 1 to N rank input operand for `argMin`, `argMax`, `batchNormalization`, `concat`, `cumulativeSum`, `gather`, `softmax` and `split` operators that take 1 axis.

@fdwr and @reillyeon  , PTAL. 

/cc @miaobin

Fix #873

***
<a href="https://pr-preview.s3.amazonaws.com/huningxin/webnn/pull/882.html" title="Last updated on Aug 27, 2025, 3:05 AM UTC (a4e0745)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/882/2c0f3a0...huningxin:a4e0745.html" title="Last updated on Aug 27, 2025, 3:05 AM UTC (a4e0745)">Diff</a>

### CODE REVIEW

**Review by fdwr — APPROVED** [review]
👍 TY NH.

I see `gatherElements` and `scatterElements` are already good.

**Review by reillyeon — APPROVED** [review]
